lib/client-assets.php

```diff
@@ -574,3 +574,50 @@ function gutenberg_register_vendor_scripts( $scripts ) {
 // Enqueue stored styles.
 add_action( 'wp_enqueue_scripts', 'gutenberg_enqueue_stored_styles' );
 add_action( 'wp_footer', 'gutenberg_enqueue_stored_styles', 1 );
+
+/**
+ * Registers interactivity scripts for Gutenberg.
+ *
+ * This function registers interactivity scripts for Gutenberg when not in the
+ * admin panel.
+ */
+function gutenberg_register_interactivity_scripts() {
+ 	if ( ! is_admin() )	 {
```

Comment:
quick note here and in `add_defer_attribute()` - another technical nit-pick

we generally prefer early-abort over nesting large code bodies, so we could rearrange these by inverting the conditions

```php
function gutenberg_register_interactivity_scripts() {
	if ( is_admin() ) {
		return;
	}

	// rest of the function
}

function add_defer_attribute( $tag, $handle ) {
	if ( ! str_starts_with( $handle, 'interactivity-' ) ) {
		return $tag;
	}

	$p = new WP_HTML_Tag_Processor( $tag );
	// rest of the code
	return $p->get_updated_html();
}
```

the value for the preference is being able to know immediately when we exit the function, not needing to continue holding on to the conditional in case the function does something after the end of the indented body